mindmeld/converter/rasa_converter.py

```diff
@@ -0,0 +1,478 @@
+                        return
+                    i = i - 1
+                file_lines.insert(insert_line, handle)
+
+    @staticmethod
+    def _attach_actions_to_function(actions, file_lines):
+        current_line = 0
+        for i, line in enumerate(file_lines):
+            if (len(re.findall("def {action}", line) > 0)):
+                current_line = i
+                break
+        while (file_lines[current_line] != ""):
+            current_line += 1
+        assert file_lines[current_line] == ""
+        file_lines[current_line:current_line] = actions
+
+    def create_init(self, mindmeld_project_directory):
+        f = self._write_init_header()
+        actions = self._read_actions()
+        templates = self._read_templates()
+        # Write all functions for each action
+        self._write_functions(actions, templates, f)
+        f.close()
+        # Read contents of current file
+        file_lines = self._read_file_lines()
+        stories_dictionary = self._get_stories()
+        # Loop through all stories and create intent-action relationship
+        for item in stories_dictionary.items():
+            # Loop through steps for each story
+            for step in item[1]:
+                # Get intent, any entities, and actions
+                intent = step['intent']
+                entities = step['entities']
+                actions = step['actions']
+                # attach handle to correct function
+                app_handle_string = RasaConverter._get_app_handle(intent, entities)
+                RasaConverter._attach_handle_to_function(app_handle_string, actions[0], file_lines)
+                # check if more than 1 action per intent
+                if len(actions) > 1:
+                    RasaConverter._attach_actions_to_function(actions[1:], file_lines)
+        # write all lines back to file
+        with open(mindmeld_project_directory + "/__init__.py", "w") as f:
+            f.writelines(file_lines)
+
+    @staticmethod
+    def create_main(mindmeld_project_directory, main_file_loc):
+        with open(main_file_loc + '/rasa_main.txt', 'r') as f:
+            string = f.read()
+        with open(mindmeld_project_directory + "/__main__.py", "w") as f:
+            f.write(string)
+
+    @staticmethod
+    def create_config(mindmeld_project_directory, main_file_loc):
+        with open(main_file_loc + '/rasa_config.txt', 'r') as f:
+            string = f.read()
+        with open(mindmeld_project_directory + "/config.py", "w") as f:
+            f.write(string)
+
+    @staticmethod
+    def create_custom_features(mindmeld_project_directory, main_file_loc):
+        with open(main_file_loc + '/rasa_custom_features.txt', 'r') as f:
+            string = f.read()
+        with open(mindmeld_project_directory + "/custom_features.py", "w") as f:
+            f.write(string)
+
+    def convert_project(self):
+        """Main function that will convert a Rasa project into a Mindmeld project
```

Comment:
We should describe more about the structure of a Rasa folder and what we can convert and the limitation